From 30290a313d8ffd46939723f390bea8b37e61a2e5 Mon Sep 17 00:00:00 2001
From: Jackie Huang <jackie.huang@windriver.com>
Date: Thu, 27 Jul 2023 15:46:40 +0800
Subject: [PATCH 11/12] Add image and pkg list for arm64

Some packages have different names for arm64,
e.g. grub-efi-amd4 vs grub-efi-arm64
and some packages doesn't support for arm64,
e.g. grub-pc, kata-container

So add debian_iso_image_arm64.inc and
debian_pkg_dirs_arm64.

Test Plan:
PASS: build-pkgs on x86-64 host
PASS: build-image on x86-64 host
PASS: build-pkgs on arm64 host
PASS: build-image on arm64 host
PASS: Deploy AIO-SX on x86-64 target
PASS: Deploy AIO-SX on arm64 target
PASS: Deploy AIO-DX on arm64 targets
PASS: Deploy std (2+2+2) on arm64 targets

Story: 2010739
Task: 47981

Depends-On: https://review.opendev.org/c/starlingx/root/+/889681
Depends-On: https://review.opendev.org/c/starlingx/root/+/889682

Change-Id: If32444fcf320b4e477f9ca3e14f593360dff162f
Signed-off-by: Jackie Huang <jackie.huang@windriver.com>
---
 debian_iso_image_arm64.inc | 427 +++++++++++++++++++++++++++++++++++++
 debian_pkg_dirs_arm64      | 106 +++++++++
 2 files changed, 533 insertions(+)
 create mode 100644 debian_iso_image_arm64.inc
 create mode 100644 debian_pkg_dirs_arm64

diff --git a/debian_iso_image_arm64.inc b/debian_iso_image_arm64.inc
new file mode 100644
index 00000000..230366d7
--- /dev/null
+++ b/debian_iso_image_arm64.inc
@@ -0,0 +1,427 @@
+# List of packages to be included/installed in ISO
+
+#base-passwd
+base-passwd
+
+#bash
+bash-builtins
+bash-dbgsym
+bash-doc
+bash-static
+bash
+
+#centos-debian-compat
+centos-debian-compat
+
+#ceph
+ceph-base
+ceph-common
+ceph-fuse
+#ceph-grafana-dashboards  # not used
+ceph-mds
+#ceph-mgr-dashboard  # not used
+#ceph-mgr-diskprediction-cloud  # not used
+#ceph-mgr-diskprediction-local  # not used
+#ceph-mgr-k8sevents  # not used
+#ceph-mgr-rook  # not used
+#ceph-mgr-ssh  # not used
+ceph-mgr
+ceph-mon
+ceph-osd
+#ceph-resource-agents  # not used
+ceph
+#cephfs-shell  # not used
+#libcephfs-dev  # not used
+#libcephfs-java  # not used
+#libcephfs-jni  # not used
+libcephfs2
+#librados-dev  # not used
+librados2
+#libradospp-dev  # not used
+#libradosstriper-dev  # not used
+libradosstriper1
+#librbd-dev  # not used
+librbd1
+#librgw-dev  # not used
+librgw2
+python3-ceph-argparse
+python3-ceph
+python3-cephfs
+python3-rados
+python3-rbd
+python3-rgw
+#rados-objclass-dev  # not used
+radosgw
+#rbd-fuse  # not used
+#rbd-mirror  # not used
+#rbd-nbd  # not used
+
+#cluster-resource-agents
+#ldirectord  # not used
+resource-agents
+
+#containerd
+containerd
+
+#crictl
+crictl
+
+#dhcp
+#isc-dhcp-client-ddns  # not used
+isc-dhcp-client
+isc-dhcp-common
+#isc-dhcp-dev  # not used
+#isc-dhcp-relay  # not used
+#isc-dhcp-server-ldap  # not used
+#isc-dhcp-server  # not used
+
+#dnsmasq
+#dnsmasq-base-lua  # not used
+dnsmasq-base
+dnsmasq-utils
+dnsmasq
+
+#docker-distribution
+docker-registry
+
+#drbd-tools
+drbd-utils
+
+#efitools
+#efitools-dbgsym  # not used
+efitools
+
+#etcd
+etcd-client
+etcd-server
+#golang-etcd-server-dev  # not used
+
+#facter
+#facter-dev  # not used
+facter
+libfacter3.14.12
+
+#gpu-operator
+#gpu-operator
+
+#grub-efi
+#grub-efi-arm64-bin  # not used
+#grub-efi-arm64-dbg  # not used
+#grub-efi-arm64-signed-template  # not used
+grub-efi-arm64
+#grub-efi  # not used
+
+#grub2
+grub-common
+#grub-coreboot-bin  # not used
+#grub-coreboot-dbg  # not used
+#grub-coreboot  # not used
+#grub-emu-dbg  # not used
+#grub-emu  # not used
+#grub-firmware-qemu  # not used
+#grub-ieee1275-bin  # not used
+#grub-ieee1275-dbg  # not used
+#grub-ieee1275  # not used
+#grub-linuxbios  # not used
+#grub-pc-bin
+#grub-pc-dbg  # not used
+#grub-pc
+#grub-rescue-pc  # not used
+#grub-theme-starfield  # not used
+#grub-xen-bin  # not used
+#grub-xen-dbg  # not used
+#grub-xen-host  # not used
+#grub-xen  # not used
+grub2-common
+#grub2
+
+#grubby
+grubby
+
+#haproxy-config
+haproxy
+haproxy-config
+
+#helm
+helm
+
+#ifupdown
+ifupdown
+ifupdown-extra
+
+kdump-tools
+
+# iputils
+iputils-arping
+iputils-ping
+iputils-tracepath
+
+#isolcpus-device-plugin
+isolcpus-device-plugin
+
+#k8s-pod-recovery
+k8s-pod-recovery
+
+#k8s-cni-cache-cleanup
+k8s-cni-cache-cleanup
+
+#kata-containers
+#kata-containers
+
+#kubectl-cert-manager
+kubectl-cert-manager
+
+# lvm2
+dmeventd
+dmsetup
+#dmsetup-udeb
+libdevmapper1.02.1
+#libdevmapper1.02.1-udeb
+#libdevmapper-dev
+libdevmapper-event1.02.1
+liblvm2cmd2.03
+#liblvm2-dev
+lvm2
+#lvm2-dbusd
+#lvm2-lockd
+#lvm2-udeb
+
+#keyrings.alt
+python3-keyrings.alt
+
+#kubernetes-x (-master, -misc, -unit-test used only for build)
+#kubernetes-1.24.4
+kubernetes-1.24.4-client
+kubernetes-1.24.4-kubeadm
+kubernetes-1.24.4-node
+#kubernetes-1.25.3
+kubernetes-1.25.3-client
+kubernetes-1.25.3-kubeadm
+kubernetes-1.25.3-node
+#kubernetes-1.26.1
+kubernetes-1.26.1-client
+kubernetes-1.26.1-kubeadm
+kubernetes-1.26.1-node
+#kubernetes-1.27.5
+kubernetes-1.27.5-client
+kubernetes-1.27.5-kubeadm
+kubernetes-1.27.5-node
+#kubernetes-1.28.4
+kubernetes-1.28.4-client
+kubernetes-1.28.4-kubeadm
+kubernetes-1.28.4-node
+#kubernetes-1.29.2
+kubernetes-1.29.2-client
+kubernetes-1.29.2-kubeadm
+kubernetes-1.29.2-node
+
+#kubernetes-unversioned
+kubernetes-unversioned
+
+#ldapscripts
+ldapscripts
+
+#libfdt
+libfdt
+
+#lighttpd
+#lighttpd-doc  # not used
+#lighttpd-mod-authn-gssapi  # not used
+#lighttpd-mod-authn-pam  # not used
+#lighttpd-mod-authn-sasl  # not used
+lighttpd-mod-geoip
+#lighttpd-mod-maxminddb  # not used
+#lighttpd-mod-trigger-b4-dl  # not used
+#lighttpd-mod-vhostdb-dbi  # not used
+#lighttpd-mod-vhostdb-pgsql  # not used
+#lighttpd-mod-webdav  # not used
+lighttpd-modules-mysql
+lighttpd
+
+#linuxptp
+linuxptp
+
+#lldpd
+#liblldpctl-dev  # not used
+lldpd
+
+#lsb
+lsb-base
+#lsb-release  # not used
+
+#net-tools
+net-tools
+
+#openldap
+ldap-utils
+libldap-2.4-2
+libldap-common
+#libldap2-dev  # not used
+#slapd-contrib  # not used
+#slapd-smbk5pwd  # not used
+slapd
+#slapi-dev  # not used
+slapd-ppolicy-check-password
+
+#parted
+#libparted-dev  # not used
+libparted-fs-resize0
+libparted-i18n
+libparted2
+parted-doc
+parted
+
+#luks-encryption
+luks-fs-mgr
+
+#pf-bb-config
+pf-bb-config
+
+#plugins
+bond-cni
+containernetworking-plugins
+#golang-github-containernetworking-plugins-dev  # not used
+
+#puppet-5.5.22
+#puppet-master-passenger  # not used
+puppet
+
+#puppet-boolean-2.0.2
+puppet-boolean
+
+#puppet-ceph-3.1.1
+puppet-module-ceph
+
+#puppet-dnsmasq
+puppet-dnsmasq
+
+#puppet-drbd-0.5.2
+puppet-drbd
+
+#puppet-hash2stuff
+puppet-hash2stuff
+
+#puppet-memcached
+puppet-memcached
+
+#puppet-etcd-1.12.3
+puppet-module-cristifalcas-etcd
+
+#puppet-keystone-17.4.0
+puppet-module-keystone
+
+#puppet-module-horizon-17.4.0
+puppet-module-horizon
+
+#puppet-ldap
+puppet-ldap
+
+#puppet-lvm-1.4.0
+puppet-lvm
+
+#puppet-network
+puppet-network
+
+#puppet-openstacklib-17.4.0
+puppet-module-openstacklib
+
+#puppet-oslo-17.4.0
+puppet-module-oslo
+
+#puppet-puppi
+puppet-puppi
+
+#puppet-rabbitmq-8.5.0
+puppet-module-puppetlabs-rabbitmq
+
+#puppet-staging
+puppet-module-nanliu-staging
+
+#puppetlabs-firewall-1.12.0
+puppet-module-puppetlabs-firewall
+
+#puppetlabs-haproxy-2.1.0
+puppet-module-puppetlabs-haproxy
+
+#puppetlabs-mysql-8.1.0
+puppet-module-puppetlabs-mysql
+
+#puppetlabs-postgresql-6.7.0
+puppet-module-puppetlabs-postgresql
+
+#puppetlabs-stdlib-5.0.0
+puppet-module-puppetlabs-stdlib
+
+#puppet-zitrlp-strongswan
+puppet-zitrlp-strongswan
+
+#python-docker
+python3-docker
+
+#python-keyring
+python3-keyring
+
+#python3-nsenter
+python3-nsenter
+
+#python3-setuptools
+#python-setuptools-doc
+python3-pkg-resources
+python3-setuptools
+
+#python3.9
+python3.9
+
+#openscap
+libopenscap25
+openscap-common
+openscap-scanner
+openscap-utils
+python3-openscap
+
+#openvswitch
+openvswitch-common
+openvswitch-switch-dpdk
+openvswitch-switch
+
+#runc
+#golang-github-opencontainers-runc-dev  # not used
+runc
+
+#shim-unsigned
+shim-helpers-arm64-signed-template
+shim-unsigned
+
+#synce4l
+synce4l
+
+#systemd
+libnss-myhostname
+#libnss-mymachines  # not used
+#libnss-resolve  # not used
+#libnss-systemd  # not used
+libpam-systemd
+#libsystemd-dev  # not used
+libsystemd0
+#libudev-dev  # not used
+libudev1
+systemd-container
+#systemd-coredump  # not used
+#systemd-journal-remote  # not used
+systemd-sysv
+#systemd-tests  # not used
+#systemd-timesyncd  # not used
+systemd
+udev
+
+#systemd-presets
+systemd-presets
+
+#trident-installer
+trident-installer
+
+#tzdata
+tzdata
+
+#watchdog
+watchdog
diff --git a/debian_pkg_dirs_arm64 b/debian_pkg_dirs_arm64
new file mode 100644
index 00000000..7f942087
--- /dev/null
+++ b/debian_pkg_dirs_arm64
@@ -0,0 +1,106 @@
+base/base-passwd
+base/cluster-resource-agents
+base/dhcp
+base/dnsmasq
+base/haproxy
+base/libfdt
+base/lighttpd
+base/linuxptp
+base/lvm2
+base/lsb
+base/pf-bb-config
+base/synce4l
+base/systemd
+base/systemd-presets
+base/watchdog
+centos-debian-compat
+ceph/ceph
+config/facter
+config/puppet-5.5.22
+config/puppet-modules/openstack/puppet-ceph-2.4.1
+config/puppet-modules/openstack/puppet-keystone-17.4.0
+config/puppet-modules/openstack/puppet-horizon-17.4.0
+config/puppet-modules/openstack/puppet-openstacklib-17.4.0
+config/puppet-modules/openstack/puppet-oslo-17.4.0
+config/puppet-modules/puppet-boolean-2.0.2
+config/puppet-modules/puppet-dnsmasq
+config/puppet-modules/puppet-drbd-0.5.2
+config/puppet-modules/puppet-etcd-1.12.3
+config/puppet-modules/puppetlabs-firewall-1.12.0
+config/puppet-modules/puppetlabs-haproxy-2.1.0
+config/puppet-modules/puppetlabs-mysql-8.1.0
+config/puppet-modules/puppetlabs-postgresql-8.0.0
+config/puppet-modules/puppetlabs-stdlib-5.0.0
+config/puppet-modules/puppet-ldap
+config/puppet-modules/puppet-hash2stuff
+config/puppet-modules/puppet-lvm-1.4.0
+config/puppet-modules/puppet-memcached
+config/puppet-modules/puppet-network
+config/puppet-modules/puppet-puppi
+config/puppet-modules/puppet-rabbitmq-8.5.0
+config/puppet-modules/puppet-staging
+config/puppet-modules/puppet-zitrlp-strongswan
+docker/python-docker
+filesystem/drbd/drbd-tools
+filesystem/luks
+filesystem/parted
+golang-github-dev/golang-github-appc-cni
+golang-github-dev/golang-github-checkpoint-restore-go-criu-dev
+golang-github-dev/golang-github-cilium-ebpf-dev
+golang-github-dev/golang-github-coreos-go-systemd-dev
+golang-github-dev/golang-github-opencontainers-specs-dev
+golang-github-dev/golang-github-vishvananda-netlink
+golang-github-dev/golang-github-networkplumbing-go-nft-dev
+golang-github-dev/golang-github-cockroachdb-datadriven
+golang-github-dev/golang-github-golang-jwt-jwt-dev
+grub/grub-efi
+#grub/grub2
+grub/grubby
+kubernetes/armada
+kubernetes/armada-helm-toolkit
+kubernetes/chartmuseum
+kubernetes/cni/bond-cni
+kubernetes/cni/plugins
+kubernetes/containerd
+kubernetes/crictl
+kubernetes/docker-distribution
+kubernetes/etcd
+kubernetes/helm
+kubernetes/k8s-cni-cache-cleanup
+kubernetes/k8s-pod-recovery
+kubernetes/kubernetes-1.24.4
+kubernetes/kubernetes-1.25.3
+kubernetes/kubernetes-1.26.1
+kubernetes/kubernetes-1.27.5
+kubernetes/kubernetes-1.28.4
+kubernetes/kubernetes-1.29.2
+kubernetes/kubernetes-unversioned
+kubernetes/plugins/isolcpus-device-plugin
+kubernetes/plugins/kubectl-cert-manager
+kubernetes/runc
+ldap/ldapscripts
+ldap/openldap
+networking/dpdk
+networking/ifupdown
+networking/ifupdown-extra
+networking/iputils
+networking/lldpd
+networking/net-tools
+networking/openvswitch
+ostree/initramfs-ostree
+ostree/mttyexec
+ostree/ostree
+ostree/ostree-upgrade-mgr
+python/dh-python
+python/python-nss
+python/python3-nsenter
+python/python3-setuptools
+python/python3.9
+python/zerorpc-python
+security/efitools
+security/keyrings.alt
+security/python-keyring
+security/shim-unsigned
+security/openscap
+storage-drivers/trident-installer
+tools/kdump-tools
-- 
2.30.2

